### ISSUE TITLE
Form Horizontal generating error when specifying columns size

### ISSUE THREAD

**Issue by danielfpedro** [Holt59/cakephp3-bootstrap-helpers#121]
<?= $this->Form->create($user, [
            'horizontal' => true,
            'columns' => [
                'md' => [
                    'label' => 2,
                    'input' => 6,
                    'error' => 4
                ]
            ]
        ]); ?>

Error
    ``Notice (8): Array to string conversion [CORE\src\View\StringTemplate.php, line 311]`` 

Removing `columns` part of the array the error ends.

**Comment by danielfpedro** [comment]
Cakephp version 3.4.3

**Comment by Holt59** [comment]
The `Form::create` option for `columns` has been dropped, you should use `setConfig('columns', [...])` prior to `create()` to get what you want. I forgot to update the documentation...

```php
$this->Form->setConfig('columns', [
    'md' => [
        'label' => 2,
        'input' => 6,
        'error' => 4
    ]
]);
echo $this->Form->create($user, ['horizontal' => true]);
```

**Comment by danielfpedro** [comment]
Awesome

**Comment by danielfpedro** [comment]
The code provided by your answer is not working... I needed to remove the `md` key to make it work but I lost the feature about set differente sizes for `sm`, `md`...

**Comment by Holt59** [comment]
I will look into this...

**Comment by Holt59** [comment]
Should be fixed by  83261d5.

**Comment by Holt59** [comment]
The previous code should actually work if you add `false` as the third argument to `setConfig`:

```php
$this->Form->setConfig('columns', [
    'md' => [
        'label' => 2,
        'input' => 6,
        'error' => 4
    ]
], false);
echo $this->Form->create($user, ['horizontal' => true]);
```

**Comment by nrother** [comment]
There is still a minor catch with a configuration like this:
```
'md' => [
    'label' => 2,
    'input' => 6,
    'error' => 4
],
'sm' => [
    'label' => 2,
    'input' => 10,
    'error' => 0
]
```
This will cause the error label to always show up under the control even on very large viewports. (In never takes it position to the right of the control with col-md-4). I'm not sure if this is a problem of Bootstrap, though.

**Comment by Holt59** [comment]
@nrother Should be fixed by a8644ed38ee22c4ad89a58b812155b1d15ab4d9c, the problem was that the `col-sm-offset-X` is effective even on large screen if no `col-md-offset-X` is present. I have added a `col-XX-offset-0` when the `error` size is not `0` for the size `XX`.

**Comment by Holt59** [comment]
I am closing this since there seem to be no problem with the latest fix. Feel free to open new issues if you have any problem.
